fdbserver/TLogServer.actor.cpp

```diff
@@ -3409,7 +3409,8 @@ ACTOR Future<Void> tLogStart(TLogData* self, InitializeTLogRequest req, Locality
 			logData->version.set(logData->unrecoveredBefore - 1);
 
 			logData->unpoppedRecoveredTags = req.allTags.size();
-			wait(initPersistentState(self, logData) || logData->removed);
+			wait(ioTimeoutError(initPersistentState(self, logData) || logData->removed,
```

Comment:
Thanks for adding this fix.

Given CC drives the state recovery machine and at-present it restarts the recovery is `sequencer` fails; it would be useful if we can add `recovery state tracking` as part of ``ClusterController::clusterWatchDatabase` loop that drives the recovery to ensure that a `recovery` doesn't get stuck in any intermediate state for a long time, if it is stuck, CC can restart `recovery` OR `start another recovery` as an _auto-healing_ mechanism. For most of the cases, restarting recovery should fix the issue; if not, then the cluster may need _manual intervention_. 

@sfc-gh-etschannen thoughts?

Reply:
it is a little tricky since there are legitimate locations recovery should get stuck. For instance if you lose a majority of coordinators or too many tlogs. Doing extra recoveries in these scenarios would be detrimental because we would could recruit useless roles as part of recovery. However, there are parts of recovery which should never stall, and in those cases I would support a cluster controller check.

Reply:
Thanks for sharing insight, Evan. I think it is worth to list the scenarios where CC progress check task could be useful. 

> For instance if you lose a majority of coordinators or too many tlogs.

For better understanding, I remember making the change where a `co-ordinator` update (`ClusterController::handleLeaderReplacement`) would lead to restart of ClusterController, hence, the recovery inflight should (or would) get restarted after the new ClusterController process gets restarted?